test/libsolidity/smtCheckerTests/inheritance/base_contract_assertion_fail_1.sol

```diff
@@ -0,0 +1,18 @@
+pragma experimental SMTChecker;
+
+contract B {
+	uint x;
+	function f() public view {
+		assert(x == 0);
+	}
+}
+
+contract C is B {
+	uint y;
+	function g() public {
+		x = 1;
+		f();
+	}
+}
+// ----
+// Warning 6328: (85-99): CHC: Assertion violation happens here.\nCounterexample:\ny = 0, x = 1\n\nTransaction trace:\nC.constructor()\nState: y = 0, x = 0\nC.g()\n    B.f() -- internal call\nState: y = 0, x = 1\nB.f()
```

Comment:
Didn't example like this already work, in the previous PR? What is special here?

Reply:
The special part here, which this PR fixes, is when an assertion fails in a base contract and the most derived contract has extra state variables.

Reply:
The bug is only related to counterexample generation, not to encoding/solving.